controller/src/beerocks/master/db/db.cpp

```diff
@@ -2721,16 +2798,102 @@ bool db::clear_client_persistent_db(const sMacAddr &mac)
     n->client_selected_bands        = beerocks::eFreqType::FREQ_UNKNOWN;
 
     LOG(DEBUG) << __FUNCTION__ << ", removing client " << mac << " from persistent db";
+    auto db_entry = client_db_entry_from_mac(mac);
+    if (!bpl::db_has_entry(type_to_string(beerocks::eType::TYPE_CLIENT), db_entry)) {
+        LOG(DEBUG) << "client entry does not exist in persistent-db for " << db_entry;
+        return true;
+    }
+
+    if (!bpl::db_remove_entry(type_to_string(beerocks::eType::TYPE_CLIENT), db_entry)) {
+        LOG(ERROR) << "failed to remove client entry " << db_entry;
+        return false;
+    }
 
     return true;
 }
 
-bool db::update_client_persistent_db(const sMacAddr &mac) { return true; }
+bool db::update_client_persistent_db(const sMacAddr &mac)
+{
+    auto n = get_node(mac);
+    if (!n) {
+        LOG(ERROR) << "node not found for mac " << mac;
+        return false;
+    } else if (n->get_type() != beerocks::TYPE_CLIENT) {
+        LOG(ERROR) << __FUNCTION__ << "node " << mac << " is not a valid client!";
+        return false;
+    }
+
+    // any persistent parameter update also sets the last-edit timestamp
+    // if it is with default value - no other persistent configuration was performed
+    if (n->client_parameters_last_edit == std::chrono::steady_clock::time_point::min()) {
+        LOG(DEBUG) << __FUNCTION__ << "persistent client parameters are empty for " << mac
+                   << ", no need to update persistent-db";
+        return true;
+    }
+
+    std::unordered_map<std::string, std::string> values_map;
+
+    //fill values map of client persistent params
+    values_map.at("timestamp") = timestamp_to_string_seconds(n->client_parameters_last_edit);
```

Comment:
All static strings (e.g. "timestamp", "timelife" etc.) should be constexpr.

Reply:
the map expects `std::string` while in c++11 it is not supported as `constexpr`
I have changed to `static const std::string` at the beginning of the `db.cpp` file

reference to the mentioned above:
https://stackoverflow.com/questions/27123306/is-it-possible-to-use-stdstring-in-a-constexpr

P.S. - I didn't use the `constexpr char constString[] = "constString";` example they suggested as the this would cause a runtime `std::string` creation from `char[]` on every use.